### PR TITLE
feat(http): call complete on request complete

### LINKED ISSUE #2635: http: should complete after response
making a `http.get().subscribe` doesn't fire `complete` after success (it should `next` only once)

```es6
observer.onNext(value);
observer.onCompleted();
```

### PR DESCRIPTION
**Pull request by PatrickJS**
closes #2635

without this it's like returning a Promise that doesn't resolve

[Review on Reviewable](https://reviewable.io/reviews/angular/angular/2666)

### CODE REVIEW

File: modules/angular2/src/http/backends/xhr_backend.ts [hunk 1]
```diff
@@ -35,9 +35,11 @@ export class XHRConnection implements Connection {
     this._xhr = new NativeConstruct();
     // TODO(jeffbcross): implement error listening/propagation
     this._xhr.open(RequestMethods[req.method], req.url);
-    this._xhr.addEventListener(
-        'load',
-        () => {this.response.onNext(new Response(this._xhr.response || this._xhr.responseText))});
+    this._xhr.addEventListener('load', () => {
+      this.response.onNext(new Response(this._xhr.response || this._xhr.responseText))
```
Comment (vicb) [inline modules/angular2/src/http/backends/xhr_backend.ts]: missing a ";" at EOL -> weird formarring

File: modules/angular2/src/http/backends/xhr_backend.ts [hunk 2]
```diff
@@ -35,9 +35,11 @@ export class XHRConnection implements Connection {
...
+          // TODO(gdi2290): defer complete if array buffer etc until done
+          this.response.onCompleted();
```
Comment (vicb) [inline modules/angular2/src/http/backends/xhr_backend.ts]: Test ?

**Comment by jeffbcross** [comment]
FYI all http commits will be blocked by #2714, which I hope to merge Monday.

**Comment by gdi2290** [comment]
yeah I saw that commit I also ran into a problem correctly testing the addEventListener handler then got distracted by AngularU. I'll update all the http commits after that's merged. If only github had a ping feature in commit messages  `ping #2666`

**Comment by gdi2290** [comment]
I have a test so this PR is good now

**Comment by jeffbcross** [comment]
Can you rebase and push @gdi2290?

**Comment by gdi2290** [comment]
@jeffbcross I rebased so the PR should be good to go when travis us green

**Comment by gdi2290** [comment]
@jeffbcross all green and ready to be merged

**Comment by jeffbcross** [comment]
@gdi2290 Thanks! I'll get it merged.

**Comment by gdi2290** [comment]
not sure why this PR is still open. merged via https://github.com/angular/angular/commit/6fac9011

**Pull request closed without merging**